File: TrafficCapture/transformationShim/src/test/java/org/opensearch/migrations/transform/shim/MatrixConfig.java
/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 */
package org.opensearch.migrations.transform.shim;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Default test matrix configuration, deserialized from {@code matrix.config.json}.
 * Test cases inherit these defaults unless they override with their own values.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
record MatrixConfig(
    List<String> defaultSolrVersions,
    String defaultOpenSearchImage
) {}

File: TrafficCapture/SolrTransformations/transforms/src/matrix.config.ts
/**
 * Test matrix configuration — single source of truth for default versions.
 *
 * Test cases inherit these defaults unless they override with their own values.
 * To add a new Solr version to the matrix, just add it here — all tests pick it up.
 */
export const matrixConfig = {
  /** Default Solr Docker image tags. Every test case runs against each of these. */
  defaultSolrVersions: ['mirror.gcr.io/library/solr:8', 'mirror.gcr.io/library/solr:9'],

  /** Default OpenSearch image for the backend. */
  defaultOpenSearchImage: 'mirror.gcr.io/opensearchproject/opensearch:3.3.0',
};
